com.sap.cloud.lm.sl.cf.core/src/test/java/com/sap/cloud/lm/sl/cf/core/util/ApplicationURITest.java

```diff
@@ -24,9 +34,100 @@
     @MethodSource("testParameters")
     public void testGetHostDomainPath(String uri, String expectedHost, String expectedDomain, String expectedPath) {
         ApplicationURI applicationURI = new ApplicationURI(uri);
-        Assertions.assertEquals(expectedDomain, applicationURI.getDomain());
-        Assertions.assertEquals(expectedDomain, ApplicationURI.getDomainFromURI(uri));
-        Assertions.assertEquals(expectedHost, applicationURI.getHost());
+        assertParametersAreValid(expectedDomain, applicationURI.getDomain(), expectedDomain, ApplicationURI.getDomainFromURI(uri),
+                                 expectedHost, applicationURI.getHost());
+        Assertions.assertEquals(expectedPath, applicationURI.getPath());
+    }
+
+    private static Stream<Arguments> routeTestParameters() {
```

Comment:
If the parameters for this parameterized test are only two, try making two test methods rather than using parameterized tests

Reply:
Refactored